### ISSUE TITLE
Error publishing desktop bridge application with MSBuild 16.10

### ISSUE THREAD

**Issue by marekdovjak** [dotnet/msbuild#6631]
Prior to MSBuild 16.10 we've been using command line to generate msixupload file 

```
nuget.exe restore My_Solution.sln

MSBuild.exe My_Solution.sln /t:My_Target_Desktop_Packaging /p:Configuration="Release" /p:AppxBundlePlatforms="x86|x64" /p:Platform="x86" /p:Platform="x64" /p:UapAppxPackageBuildMode=StoreUpload /p:AppxPackageDir="./../AppPackages" /p:PackageCertificateKeyFile="distribution-cert.pfx" /p:PackageCertificatePassword=*** /v:minimal /nr:false
```

Everything was working good. After upgrading to MSBuild 16.10, msixupload file is still generated, but x64 version of application stopped working. If you launch the application, nothing happens. App exits with code -2147450750

If I swap x86,x64 to x64,x86. Then x64 works, but x86 does not
If I clean my solution (git clean -xdf) and publish only x64, it works
If I clean my solution (git clean -xdf) and publish only x86, it works
If I clean my solution (git clean -xdf) and publish x86, it works, then I publish x64 and it doesn't work

So it looks like something from previous platform build impacts next platform build and causes it to not work. After downgrading to MSBuild 16.9, everything works.

**Comment by rainersigwald** [comment]
This sounds like a problem with the packaging logic, which is not part of this repo. Can you please:

1. Confirm that this still reproduces with 16.10.3 (we fixed a bug with solution `/t:Something` in 16.10 patches, and you may be seeing a symptom of that).
2. If it still reproduces, [open a feedback ticket instead](https://docs.microsoft.com/visualstudio/ide/feedback-options)?

After it's created, that will open an internal bug which will be routed to the right team. If you post the link here we may be able to help that process go faster.

**Comment by marekdovjak** [comment]
1. Yes, problem still reproduces with 16.10.3 (also with 2022 preview). We also had issue with `/t:Something`, but it was fixed. 
2. https://developercommunity.visualstudio.com/t/Error-publishing-desktop-bridge-applicat/1467192
